PowerValidatedSolutions.psm1

```diff
@@ -50066,6 +50066,372 @@ Function Test-WMSubnetInput {
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$certDir,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$commonName,
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$subjectAlternativeNames,
+        [Parameter (Mandatory = $false)] [ValidateSet (2048, 3072, 4096)] [int]$keySize = 2048,
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [int]$expireDays = 365,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$organization,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$organizationUnit,            
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$locality,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$state,
+        [Parameter (Mandatory = $true)] [ValidateSet ("US", "CA", "AX", "AD", "AE", "AF", "AG", "AI", "AL", "AM", "AN", "AO", "AQ", "AR", "AS", "AT", "AU", `
+        "AW", "AZ", "BA", "BB", "BD", "BE", "BF", "BG", "BH", "BI", "BJ", "BM", "BN", "BO", "BR", "BS", "BT", "BV", "BW", "BZ", "CA", "CC", "CF", "CH", "CI", "CK", `
+        "CL", "CM", "CN", "CO", "CR", "CS", "CV", "CX", "CY", "CZ", "DE", "DJ", "DK", "DM", "DO", "DZ", "EC", "EE", "EG", "EH", "ER", "ES", "ET", "FI", "FJ", "FK", `
+        "FM", "FO", "FR", "FX", "GA", "GB", "GD", "GE", "GF", "GG", "GH", "GI", "GL", "GM", "GN", "GP", "GQ", "GR", "GS", "GT", "GU", "GW", "GY", "HK", "HM", "HN", `
+        "HR", "HT", "HU", "ID", "IE", "IL", "IM", "IN", "IO", "IS", "IT", "JE", "JM", "JO", "JP", "KE", "KG", "KH", "KI", "KM", "KN", "KR", "KW", "KY", "KZ", "LA", `
+        "LC", "LI", "LK", "LS", "LT", "LU", "LV", "LY", "MA", "MC", "MD", "ME", "MG", "MH", "MK", "ML", "MM", "MN", "MO", "MP", "MQ", "MR", "MS", "MT", "MU", "MV", `
+        "MW", "MX", "MY", "MZ", "NA", "NC", "NE", "NF", "NG", "NI", "NL", "NO", "NP", "NR", "NT", "NU", "NZ", "OM", "PA", "PE", "PF", "PG", "PH", "PK", "PL", "PM", `
+        "PN", "PR", "PS", "PT", "PW", "PY", "QA", "RE", "RO", "RS", "RU", "RW", "SA", "SB", "SC", "SE", "SG", "SH", "SI", "SJ", "SK", "SL", "SM", "SN", "SR", "ST", `
+        "SU", "SV", "SZ", "TC", "TD", "TF", "TG", "TH", "TJ", "TK", "TM", "TN", "TO", "TP", "TR", "TT", "TV", "TW", "TZ", "UA", "UG", "UM", "US", "UY", "UZ", "VA", `
+        "VC", "VE", "VG", "VI", "VN", "VU", "WF", "WS", "YE", "YT", "ZA", "ZM", "COM", "EDU", "GOV", "INT", "MIL", "NET", "ORG", "ARPA")] [String]$country
+    )
+
+    # Validate inputs
+    if (-Not (Test-Path -Path $certDir)) {
+        Write-Error "Certificate directory $certDir does not exist: PRE_VALIDATION_FAILED" -ErrorAction Stop
+    }
+    if (-Not ([bool] (Get-Command -ErrorAction Ignore -Type Application openssl))) {
+        Write-Error "OpenSSL not found.  Please verify OpenSSLis installed: PRE_VALIDATION_FAILED" -ErrorAction Stop
+    } else {
+        $opensslVersion = openssl version
```

Comment:
```suggestion
        $opensslVersion = openssl version | ForEach-Object { ($_ -split ' ')[1] }
```

This would get the exact version.

```powershell
PS > openssl version | ForEach-Object { ($_ -split ' ')[1] }
3.2.1
```

Reply:
addressed in commit a1d3ac659124752299f686a91a245fc9dcaf00b5